packages/financial-templates-lib/src/price-feed/TraderMadePriceFeed.js

```diff
@@ -0,0 +1,278 @@
+
+    // 1. Construct URLs.
+    const priceUrl = `https://marketdata.tradermade.com/api/v1/live?currency=${this.pair}&api_key=${this.apiKey}`;
+    const ohlcMinuteUrl = `https://marketdata.tradermade.com/api/v1/timeseries?currency=${this.pair}&api_key=${this.apiKey}&start_date=${startDate}&end_date=${endDate}&format=records&interval=minute&period=${this.ohlcPeriod}`;
+    const ohlcHourlyUrl = `https://marketdata.tradermade.com/api/v1/timeseries?currency=${this.pair}&api_key=${this.apiKey}&start_date=${startDate}&end_date=${endDate}&format=records&interval=hourly`;
+
+    // 2. Send requests.
+    const [priceResponse, ohlcMinuteResponse, ohlcHourlyResponse] = await Promise.all([
+      this.networker.getJson(priceUrl),
+      this.networker.getJson(ohlcMinuteUrl),
+      this.networker.getJson(ohlcHourlyUrl)
+    ]);
+
+    // 3. Check responses.
+    if (!priceResponse || !priceResponse.quotes || !priceResponse.quotes[0].ask) {
+      throw new Error(`🚨Could not parse price result from url ${priceUrl}: ${JSON.stringify(priceResponse)}`);
+    }
+
+    if (!ohlcMinuteResponse || !ohlcMinuteResponse.quotes || !ohlcMinuteResponse.quotes[0].close) {
+      throw new Error(
+        `🚨Could not parse ohlc minute price result from url ${ohlcMinuteUrl}: ${JSON.stringify(ohlcMinuteResponse)}`
+      );
+    }
+
+    if (!ohlcHourlyResponse || !ohlcHourlyResponse.quotes || !ohlcHourlyResponse.quotes[0].close) {
+      throw new Error(
+        `🚨Could not parse ohlc hourly price result from url ${ohlcHourlyUrl}: ${JSON.stringify(ohlcHourlyResponse)}`
+      );
+    }
+
+    // 4. Parse results.
+    // Return data structure:
+    //  {
+    //     "endpoint": "live",
+    //     "quotes": [
+    //       {
+    //         "ask": 0.15431,
+    //         "base_currency": "CNY",
+    //         "bid": 0.15431,
+    //         "mid": 0.15431,
+    //         "quote_currency": "USD"
+    //       }
+    //     ],
+    //     "requested_time": "Tue, 26 Jan 2021 01:29:52 GMT",
+    //     "timestamp": 1611624593
+    //   }
+    // For more info, see: https://marketdata.tradermade.com/documentation
+    const newPrice = this.convertPriceFeedDecimals(priceResponse.quotes[0].ask);
+
+    // Return data structure:
+    // {
+    //   "base_currency": "CNY",
+    //   "end_date": "2021-01-26 03:31:00",
+    //   "endpoint": "timeseries",
+    //   "quote_currency": "USD",
+    //   "quotes": [
+    //     {
+    //       "close": 0.1543,
+    //       "date": "2021-01-26 00:00:00",
+    //       "high": 0.1543,
+    //       "low": 0.1543,
+    //       "open": 0.1543
+    //     },
+    //     ...
+    //   ]
+    //   "request_time": "Wed, 27 Jan 2021 01:45:28 GMT",
+    //   "start_date": "2021-01-26-00:01"
+    // }
+    // For more info, see: https://marketdata.tradermade.com/documentation
+    const newHistoricalPricesMinute = ohlcMinuteResponse.quotes
+      .map(ohlcMinute => ({
```

Comment:
So it seems we don't care about the `open` , `high`, or `low` prices right?

Reply:
yes, only use close price